### PR TITLE
Add Execution Duration metadata for dbt tests

### PR DESCRIPTION
**Pull request by johannkm**
title

### CODE REVIEW

**Comment by johannkm** [comment]
* **#22008** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22008?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @johannkm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py [hunk 1]
```diff
@@ -541,8 +545,22 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
             ),
         ]
 
+        non_deterministic_metadata_keys = ["Execution Duration"]
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_checks.py:548]: Add a comment that we're removing this for tests?

File: python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py [hunk 2]
```diff
@@ -296,7 +298,6 @@ def to_default_asset_events(
                     output_name=dagster_name_fn(event_node_info),
                     metadata={
                         **default_metadata,
-                        "Execution Duration": duration_seconds,
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py:299]: Do we have a test that ensures that this is still present on `AssetMaterialization` and `Output`?

**Review by rexledesma — APPROVED** [review]
Looks fine! If a test doesn't already exist, please add a test on `Execution Duration` for `AssetMaterialization` and `Output` as well!

Just want to make sure we have confidence about refactoring the metadata keys in the future 🙏

**Pull request merged**